store/tikv/region_cache.go

```diff
@@ -69,71 +69,71 @@ type Region struct {
 	lastAccess int64          // last region access time, see checkRegionCacheTTL
 }
 
+// EngineCnt presents max engine type supported by TiDB.
+const EngineCnt = 2
+
 // RegionStore represents region stores info
 // it will be store as unsafe.Pointer and be load at once
 type RegionStore struct {
-	workTiKVIdx    int32    // point to current work peer in meta.Peers and work store in stores(same idx) for tikv peer
-	workTiFlashIdx int32    // point to current work peer in meta.Peers and work store in stores(same idx) for tiflash peer
-	stores         []*Store // stores in this region
-	storeFails     []uint32 // snapshots of store's fail, need reload when `storeFails[curr] != stores[cur].fail`
+	currIdx    [EngineCnt]int32
+	stores     [EngineCnt][]*Store // stores in this region
+	storeFails [EngineCnt][]uint32 // snapshots of store's fail, need reload when `storeFails[curr] != stores[cur].fail`
 }
 
 // clone clones region store struct.
 func (r *RegionStore) clone() *RegionStore {
-	storeFails := make([]uint32, len(r.stores))
-	for i, e := range r.storeFails {
-		storeFails[i] = e
+	nr := &RegionStore{
+		currIdx: r.currIdx,
 	}
-	return &RegionStore{
-		workTiFlashIdx: r.workTiFlashIdx,
-		workTiKVIdx:    r.workTiKVIdx,
-		stores:         r.stores,
-		storeFails:     storeFails,
+	for engineType := range r.stores {
```

Comment:
Why can't be `nr.stores[0] = r.stores[0]`?

Reply:
no, this need deep copy

Reply:
In the old code it's not deep copy. So the old code is a bug?

Reply:
ok, I will change to `nr.stores[i] = r.stores[i]`